### PR TITLE
PHP 7 and PostgreSQL 10 compatibility

### PR DESCRIPTION
**Pull request by y92**
In this code, I found several classes where the constructor function had the same name than the class. That is deprecated in PHP 7, and when I test this version on my machine, I get warning messages. The good way to define a constructor is to use the name __construct for the constructor.

Furthermore, the version 10 of PostgreSQL was released few months ago, and phpPgAdmin was not compatible with this new version of PgSQL, so I corrected this problem.

function each() is not compatible with PHP 7.2, which was released few days ago, so I also corrected this.

PS : Sorry if I submit my pull request again, but I had some problems with my old repository and I had to create it again.

### CODE REVIEW

File: classes/database/Connection.php [hunk 1]
```diff
@@ -74,31 +74,36 @@ function getDriver(&$description) {
...
+	    switch (substr($version,0,2)) {
+		case '10': return 'Postgres10'; break;
+	    }
 
-		// If unknown version, then default to latest driver
-		return 'Postgres';
+	    /* All <7.4 versions are not supported */
+	    // if major version is 7 or less and wasn't cought in the
+	    // switch/case block, we have an unsupported version.
+	    if ((int)substr($version, 0, 1) < 8 && (int) substr($version,0,2) < 10) {
```
Comment (nirgal) [inline classes/database/Connection.php]: I'm using
```
if ((int)strtok($version, '.') < 8)
```
to support major releases with 2 digits.

**Comment by DeyV** [comment]
It's look OK, and we are waiting, waiting for this pullrequest

**Comment by tangingw** [comment]
@y92 
I hacked phpPGAdmin-5.1 and added the modified based on 942d054. It is able to read my PostgreSQL version 10 server. Thanks for the modification.

**Comment by tfl** [comment]
Thanks for the work, I have a 11.1 instance running and to view the functions a small change was necessary. See attached diff.
[functions.diff.txt](https://github.com/phppgadmin/phppgadmin/files/2724385/functions.diff.txt)

**Comment by xzilla** [comment]
> I'd love to get a clean patch (ie. no whitespace changes, no extra version designation) with the "each" fixes in this specific commit as its own feature branch PR against current master. If you're not up for that I'd understand, and I'll probably manually pull these changes in before next release. Thanks for your original submission!

We actually did receive a separate PR for the "each" fixes, which I have gone ahead and pulled in. I think that addresses all the issues in this PR. If you have additional fixes or features, please re-submit as separated PR's per the DEVELOPERS file.  Thanks again for your interest & efforts!

**Pull request closed without merging**